File: learn/learn-algo/leetcode/1154/go/1154.go
package main

import (
	"fmt"
	"time"
)

func dayOfYear(date string) int {
	t, _ := time.Parse("2006-01-02", date)
	s, _ := time.Parse("2006-01-02", date[:4]+"-01-01")

	res := t.Sub(s).Hours()/24 + 1

	fmt.Println(t)
	fmt.Println(s)

	return int(res + 1)
}

func main() {
	res := dayOfYear("2021-12-24")
	fmt.Println(res)
}

File: learn/learn-algo/leetcode/1154/c++/1154.cpp
#include <iostream>
#include <string>
#include <vector>
#include <algorithm>
#include <numeric>
#include <math.h>
#include <time.h>
#include <stdlib.h>
#include <ctime>
#include <iomanip>
#include <sstream>


using namespace std;

class Solution {
public:
    int dayOfYear(string date) {
        tm t{};
        t.tm_year = stoi(date.substr(0, 4)) - 1900;
        t.tm_mon = stoi(date.substr(5, 2)) - 1;
        t.tm_mday = stoi(date.substr(8, 2));
        mktime(&t);
        return t.tm_yday + 1;
    }

    int dayOfYear2(string date) {
        tm t{};
        istringstream ss(date);
        ss >> get_time(&t, "%Y-%m-%d");
        mktime(&t);
        return t.tm_yday + 1;
    }
};


int main() {
    Solution s;
    auto res = s.dayOfYear2("2019-01-09");
    cout << res << endl;
    return 0;
}